lib/phoenix_test/playwright/browser_context.ex

```diff
@@ -18,6 +20,33 @@ defmodule PhoenixTest.Playwright.BrowserContext do
     resp.result.page.guid
   end
 
+  @doc """
+  Add cookies to the browser context.
+
+  See `PhoenixTest.Playwright.Cookies` for the "shape" of the cookie map.
+  """
+  def add_cookies(context_id, cookies) do
+    cookies = Enum.map(cookies, &Cookies.to_params_map/1)
+    post(guid: context_id, method: :add_cookies, params: %{cookies: cookies})
+  end
+
+  @doc """
+  Add a `Plug.Session` cookie to the browser context.
+
+  This is useful for emulating a logged-in user.
+
+  Note that that the cookie `:value` must be a map, since we are using
+  `Plug.Conn.put_session/3` to write each of value's key-value pairs
+  to the cookie.
+
+  The `session_options` are exactly the same as the opts used when
+  writing `plug Plug.Session` in your router/endpoint module.
+  """
+  def add_session_cookie(context_id, cookie, session_options) do
+    cookie = Cookies.to_session_params_map(cookie, session_options)
+    post(guid: context_id, method: :add_cookies, params: %{cookies: [cookie]})
+  end
```

Comment:
How about removing this from `BrowserContext` and inlining in `PhoenixTest.Playwright`?

That way, we can limit `BrowserContext` to being a slim wrapper of the playwright API.

Reply:
I am happy to do that. I do not mind where it lives.

In case it influences your thinking (it did mine), consider the playwright docs:
https://playwright.dev/docs/api/class-browsercontext

Since `addCookies` is a method on the BrowserContext, it seemed right to put it in that module. Then `PhoenixTest.Playwright` allows devs to use the nice "pipe" flow (thanks to `tap/2`).

Reply:
I'm definitely in favour of `BrowserContext.add_cookies` staying where it is.
Just moving the phoenix/plug-specific `add_session_cookie` logic.

Reply:
I agree!